src/include/sys-zlib.h

```diff
@@ -732,7 +732,7 @@ extern z_const char * const z_errmsg[10]; /* indexed by 2-zlib_error */
 #  define OS_CODE  10
 #endif
 
-#ifdef _BEOS_
+#if defined _BEOS_ || defined __HAIKU__
```

Comment:
The extraction for %sys-zlib.h and %u-zlib.c is actually done via a script that runs on the official zlib sources:

  https://github.com/metaeducation/ren-c/blob/master/tools/make-zlib.r

Since the last capture was done it looks like they had a PR which added in some `__HAIKU__` bits, but not this OS_CODE part:

  https://github.com/madler/zlib/pull/830/files

That was very recent (July 29).

If this change is correct, then sending it as a PR to zlib would be ideal.  Then I'd just re-run a capture, picking up any other potentially important patches.